### ISSUE TITLE
Nonblocking search using async

### ISSUE THREAD

**Issue by Benjamin-Lee** [krisk/Fuse#560]
### Description
Right now, it appears that adding search functionality blocks the UI thread while it runs. While I can move it to a webworker, it would be ideal to just await the results.

### Describe the solution you'd like
Use async js when searching a large (10k element) JSON array.

### Describe alternatives you've considered
Webworkers?

**Comment by cschweda** [comment]
+1

Just noticing this now. UI freezes as search is generating.

**Comment by cschweda** [comment]
+1  -- any ideas on this?

**Comment by itsadamlove** [comment]
+1 also having issues with this

**Comment by prithvi2k2** [comment]
@krisk is this blocking of io addressed?

**Comment by krisk** [comment]
JavaScript is single-threaded — making `search()` async doesn't make it non-blocking. The computation still has to happen somewhere.

If search is blocking your UI, the real options are:

1. **Web Worker** — run Fuse in a worker to keep the main thread free. This is a wrapper you set up in your app, not something Fuse should own (worker lifecycle, serialization, message passing are all app-level concerns).
2. **Reduce the dataset** — pre-filter before searching, or use `limit` to enable the heap-based fast path.
3. **Debounce** — don't search on every keystroke.

Fuse is a client-side library for datasets that fit in memory and search in milliseconds. If you're searching millions of records, a server-side engine is the right tool.

**Comment by krisk** [comment]
Well, turns out Web Workers aren't overkill after all. I take it back.

I've been working on built-in parallel search support via a new `FuseWorker` class. Instead of running search on the main thread (which blocks the UI), `FuseWorker` splits your dataset across multiple Web Workers and searches them in parallel.

**Usage:**
```js
import { FuseWorker } from 'fuse.js/worker'

const fuse = new FuseWorker(docs, options)
const results = await fuse.search('query')
fuse.terminate()
```

Same options, same results as `Fuse` — just async. Workers are created lazily on first search, no setup needed.

**Benchmarks (100K documents, browser):**

| Method | Time | Speedup |
|---|---|---|
| Fuse (single thread) | 6,276ms | 1.00x |
| FuseWorker (2 workers) | 3,514ms | 1.79x |
| FuseWorker (4 workers) | 1,909ms | 3.29x |
| FuseWorker (8 workers) | 1,338ms | 4.69x |

More importantly, the UI stays completely responsive during search — no jank, no frozen frames.

The API is intentionally small: `search`, `add`, `setCollection`, `terminate`. If you need to remove items, call `setCollection` with a filtered array.

Branch: [`feat/web-workers`](https://github.com/krisk/Fuse/tree/feat/web-workers) | Commit: [`6e54a1d4`](https://github.com/krisk/Fuse/commit/6e54a1d4)

Planning to ship this as a beta release. Feedback welcome — especially on the API surface and whether the worker count defaults make sense for your use cases.

**Comment by krisk** [comment]
This is now published as `7.4.0-beta.1`:

```bash
npm install fuse.js@beta
```

Docs and interactive demo: [fusejs.io/web-workers.html](https://www.fusejs.io/web-workers.html)

Feedback welcome — especially on the API surface and real-world performance on your datasets.
